Library/Formula/node.rb

```diff
@@ -73,6 +73,10 @@ def install
       cd buildpath/"npm_install" do
         system "./configure", "--prefix=#{libexec}/npm"
         system "make", "install"
+        # `package.json` has relative paths to the npm_install directory.
+        # This copies back over the vanilla `package.json` that is expected.
+        # https://github.com/Homebrew/homebrew/issues/46131#issuecomment-157845008
+        cp buildpath/"npm_install/package.json", libexec/"npm/lib/node_modules/npm"
```

Comment:
Ah, yes, sorry, I see now. Yeh, this approach seems :+1: to me but will let @DomT4 give the final :ok:.